cdap-app-fabric/src/main/java/io/cdap/cdap/app/DefaultAppConfigurer.java

```diff
@@ -112,6 +120,7 @@ public DefaultAppConfigurer(Id.Namespace namespace, Id.Artifact artifactId, Appl
     this.pluginInstantiator = pluginInstantiator;
     this.appMetadata = new HashMap<>();
     this.triggerFactory = new DefaultTriggerFactory(namespace.toEntityId());
+    this.featureFlags = featureFlags;
```

Comment:
Do a defensive copy and make it immutable.

`Collections.unmodifiableMap(new HashMap<>(featureFlags));`